internal/access/credential.go

```diff
@@ -73,3 +79,71 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 
 	return nil
 }
+
+func lowercaseCheck(n int, password string) bool {
+	match, _ := regexp.MatchString(fmt.Sprintf(`(.*\p{Ll}){%d,}`, n), password)
+	fmt.Println(match)
+	return match
+}
+
+func uppercaseCheck(n int, password string) bool {
+	match, _ := regexp.MatchString(fmt.Sprintf(`(.*\p{Lu}){%d,}`, n), password)
+	return match
+}
+
+func numberCheck(n int, password string) bool {
+	match, _ := regexp.MatchString(fmt.Sprintf(`(.*\p{N}){%d,}`, n), password)
+	return match
+}
+
+// list is from OWASP, wikipedia
+func symbolCheck(n int, password string) bool {
+	match, _ := regexp.MatchString(fmt.Sprintf(`(.*[ !"#$%%&'()*+,-./\:;<=>?@^_{}|~%s%s]){%d,}`, regexp.QuoteMeta(`/\[]`), "`", n), password)
+	return match
+	// /\[]`
+}
+
+func checkPasswordRequirements(password string) error {
```

Comment:
One requirement that isn't demonstrated here is the NIST password recommendations:

* [minimum 8 character length without additional complexity requirements](https://pages.nist.gov/800-63-3/sp800-63b.html#5111-memorized-secret-authenticators)
* [check passwords against a corpus of breached passwords and a list of common words/passwords.](https://pages.nist.gov/800-63-3/sp800-63b.html#5112-memorized-secret-verifiers)

The second recommendation is missing and I don't a good way of integrating with this implementation